### DISCUSSION TITLE
Error from nvim-dap-virtual-text

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [LazyVim/LazyVim#821] [Q&A] [answered]
Has anyone seen this `nvim-dap-virtual-text` error and have a suggestion on how to fix/mitigate it?

![image](https://github.com/LazyVim/LazyVim/assets/994357/282df24a-8ea7-4685-a214-4db6aa0b0d27)

```
Error executing vim.schedule lua callback: ...-virtual-text/lua/nvim-dap-virtual-text/virtual_text.lua:213: Invalid 'virt_text_pos'
stack traceback:
	[C]: in function 'nvim_buf_set_extmark'
	...-virtual-text/lua/nvim-dap-virtual-text/virtual_text.lua:213: in function 'set_virtual_text'
	...lazy/nvim-dap-virtual-text/lua/nvim-dap-virtual-text.lua:112: in function 'c'
	...k/.local/share/LazyVim/lazy/nvim-dap/lua/dap/session.lua:953: in function <...k/.local/share/LazyVim/lazy/nvim-dap/lua/dap/session.lua:947>
```

### Background

I wanted to enable python debugging, so I added the [LazyVim DAP core extras](https://www.lazyvim.org/plugins/extras/dap.core) to `lazy.lua` and figured I'd build on top of that. I then added my own nvim-dap python debugging config.

The related config files:
- LazyVim config: https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/config/lazy.lua#L21
- debug config: https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/plugins/debug.lua
- keymap for running test with DAP: https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/plugins/test.lua

I can debug python tests just fine, but everytime I step forward, I get a _pair_ of errors about the `display_callback` defined in the [`nvim-dap-virtual-text`](https://github.com/theHamsta/nvim-dap-virtual-text) default options.

I can get rid of the error by overriding the default `display_callback`:

```lua
{
    "theHamsta/nvim-dap-virtual-text",
    config = function()
        require("nvim-dap-virtual-text").setup({
            display_callback = function(variable, buf, stackframe, node, options) end,
        })
    end,
},

```

**Comment by folke** [comment]
You should report this in the nvin-dap-virtual-text repo. Probably related to the changes for inline virtual text.

Please do report an issue there, otherwise the maintainer can;t know something is wrong and they can't fix it.

**Reply by fredrikaverpil** [reply]
Ok, thanks for the input, I made a report [here](https://github.com/theHamsta/nvim-dap-virtual-text/issues/61).

**Reply by fredrikaverpil** [reply]
Like you mentioned, yeah, when reverting back to an earlier commit https://github.com/theHamsta/nvim-dap-virtual-text/commit/10f858c402eb985739627b7e69a38f5b2d79076b the crash isn't happening.

**Comment by fredrikaverpil** [comment, marked as answer]
I was using the nightly build of neovim (unintentionally). By using the non-nightly neovim, the issues vanished.
